src/Libraries/Microsoft.Extensions.DataIngestion.MarkItDown/MarkItDownMcpReader.cs

```diff
@@ -0,0 +1,125 @@
+    {
+        _ = Throw.IfNull(source);
+        _ = Throw.IfNullOrEmpty(identifier);
+
+        if (!source.Exists)
+        {
+            throw new FileNotFoundException("The specified file does not exist.", source.FullName);
+        }
+
+        // Read file content as base64 data URI
+#if NET
+        byte[] fileBytes = await File.ReadAllBytesAsync(source.FullName, cancellationToken).ConfigureAwait(false);
+#else
+        byte[] fileBytes;
+        using (FileStream fs = new(source.FullName, FileMode.Open, FileAccess.Read, FileShare.Read))
+        {
+            using MemoryStream ms = new();
+            await fs.CopyToAsync(ms).ConfigureAwait(false);
+            fileBytes = ms.ToArray();
+        }
+#endif
+        string base64Content = Convert.ToBase64String(fileBytes);
+        string mimeType = string.IsNullOrEmpty(mediaType) ? "application/octet-stream" : mediaType!;
+        string dataUri = $"data:{mimeType};base64,{base64Content}";
+
+        string markdown = await ConvertToMarkdownAsync(dataUri, cancellationToken).ConfigureAwait(false);
+
+        return MarkdownParser.Parse(markdown, identifier);
+    }
+
+    /// <inheritdoc/>
+    public override async Task<IngestionDocument> ReadAsync(Stream source, string identifier, string mediaType, CancellationToken cancellationToken = default)
+    {
+        _ = Throw.IfNull(source);
+        _ = Throw.IfNullOrEmpty(identifier);
+
+        // Read stream content as base64 data URI
+        using MemoryStream ms = new();
+#if NET
+        await source.CopyToAsync(ms, cancellationToken).ConfigureAwait(false);
+#else
+        await source.CopyToAsync(ms).ConfigureAwait(false);
+#endif
+        byte[] fileBytes = ms.ToArray();
+        string base64Content = Convert.ToBase64String(fileBytes);
+        string mimeType = string.IsNullOrEmpty(mediaType) ? "application/octet-stream" : mediaType;
+        string dataUri = $"data:{mimeType};base64,{base64Content}";
+
+        string markdown = await ConvertToMarkdownAsync(dataUri, cancellationToken).ConfigureAwait(false);
+
+        return MarkdownParser.Parse(markdown, identifier);
+    }
+
+    private async Task<string> ConvertToMarkdownAsync(string dataUri, CancellationToken cancellationToken)
+    {
+        // Create HTTP client transport for MCP
+#pragma warning disable CA2007 // Consider calling ConfigureAwait on the awaited task - await using pattern
+        await using var transport = new HttpClientTransport(new HttpClientTransportOptions
+        {
+            Endpoint = _mcpServerUri
+        });
+#pragma warning restore CA2007
+
+        // Create MCP client
+#pragma warning disable CA2007 // Consider calling ConfigureAwait on the awaited task
+        await using var client = await McpClient.CreateAsync(transport, cancellationToken: cancellationToken).ConfigureAwait(false);
+#pragma warning restore CA2007 // Consider calling ConfigureAwait on the awaited task
```

Comment:
These pragmas seem to be unnecessary? 

```suggestion
        await using var transport = new HttpClientTransport(new HttpClientTransportOptions
        {
            Endpoint = _mcpServerUri
        });

        // Create MCP client
        await using var client = await McpClient.CreateAsync(transport, cancellationToken: cancellationToken).ConfigureAwait(false);
```